deepchem/models/jax_models/jax_model.py

```diff
@@ -0,0 +1,371 @@
+import numpy as np
+import time
+import logging
+try:
+  from collections.abc import Sequence as SequenceCollection
+except:
+  from collections import Sequence as SequenceCollection
+
+from deepchem.data import Dataset
+from deepchem.models.models import Model
+from deepchem.models.losses import Loss
+from deepchem.models.optimizers import Optimizer
+from typing import Any, Callable, Iterable, List, Optional, Tuple, Union
+from deepchem.utils.typing import LossFn
+
+# JAX depend
+import jax.numpy as jnp
+import jax
+import haiku as hk
+import optax
+
+import warnings
+
+logger = logging.getLogger(__name__)
+
+
+class JaxModel(Model):
+  """This is a DeepChem model implemented by a Jax Model
+
+  Here is a simple example of that uses JaxModel to train a
+  Haiku (JAX Neural Network Library) based model on deepchem
+  dataset.
+
+  >> def f(x):
```

Comment:
Since you're using `>>` instead of `>>>` this test won't run during doctests. Is that on purpose?

Reply:
Yes it was intentional, doctest are run in `deepchem core` workflow where I haven't included `jax` dependency.

Reply:
I think, now that we have mostly sorted out the CI, i think i can add `Jax` in here https://github.com/deepchem/deepchem/blob/master/env.common.yml

Reply:
Sounds reasonable! We shouldn't add Jax to common though I think since that's the platform free core for DeepChem